### PR TITLE
feat(docs, cli): CLI auth

### PR DESCRIPTION
**Pull request by Strajk**
https://github.com/e2b-dev/e2b/assets/697301/9adc4192-cc6d-42dd-bc84-caafd973699b

### CODE REVIEW

File: packages/cli/src/commands/auth.ts [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+
+export function getUserConfig(): UserConfig | null {
+  if (!fs.existsSync(USER_CONFIG_PATH)) return null
+  return JSON.parse(fs.readFileSync(USER_CONFIG_PATH, 'utf8'))
+}
+
+async function signInWithBrowser(): Promise<UserConfig> {
+  const server = http.createServer()
+  const { port } = await listen(server, 0, '127.0.0.1')
+  const loginUrl = new URL(DOCS_BASE + '/docs/api/cli')
```
Comment (jakubno) [inline packages/cli/src/commands/auth.ts]: Double `docs`, made me remember`log/blog` :D 
```suggestion
  const loginUrl = new URL(DOCS_BASE + '/api/cli')
```

File: packages/cli/src/commands/auth.ts [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+  loginUrl.searchParams.set('next', `http://localhost:${port}`)
+  loginUrl.searchParams.set('cliVersion', pkg.version)
+
+  return new Promise((resolve, reject) => {
+    server.once('request', (req, res) => {
+      // Close the HTTP connection to prevent `server.close()` from hanging
+      res.setHeader('connection', 'close')
+      const followUpUrl = new URL(DOCS_BASE + '/docs/api/cli')
```
Comment (jakubno) [inline packages/cli/src/commands/auth.ts]:
```suggestion
      const followUpUrl = new URL(DOCS_BASE + '/api/cli')
```

**Review by jakubno — DISMISSED** [review]
There's still a lot of devbook reference/ devbook sdk imports. We don't have to do it here, but I think we should clean it up (at least readme).

File: packages/cli/src/commands/auth.ts (hunk 2 above)
Comment (Strajk) [reply packages/cli/src/commands/auth.ts]: danke!

**Pull request merged**